edx_rbac/mixins.py

```diff
@@ -61,3 +64,128 @@ def check_permissions(self, request):
             self.permission_denied(
                 request,
                 message=('MISSING: {}'.format(', '.join(missing_permissions))))
+
+
+class PermissionRequiredForListingMixin(PermissionRequiredMixin):
+    """
+    Mixin for checking role-based access against multiple objects, specifically
+    using the "list" action of a DRF ViewSet.  The brunt of the work is done in
+    the `accessible_contexts` cached property - this property determines the set
+    of "contexts" (probably identifiers of some model) to which the requesting
+    user has access under any of the `allowed_roles`.  It's pretty opinionated and
+    you _must_ define the following fields:
+
+    `list_lookup_field` - The field of the model you want to list instances of
+    that holds values defined by `accessible_contexts`.
```

Comment:
Nit: Might be nice to add that it expects a string to address Adam's comment, because some might be used to Django's model linking where you can use a string or module path